pkg/providers/ovirt/mapper/mapper.go

```diff
@@ -206,5 +283,158 @@ func (o *OvirtMapper) mapCPU() *kubevirtv1.CPU {
 		}
 	}
 
+	// Custom cpu model
+	if customCPU, available := o.vm.CustomCpuModel(); available {
+		cpu.Model = customCPU
+	}
+
 	return &cpu
 }
+
+func (o *OvirtMapper) mapHighAvailability() *bool {
+	enabled := false
+	ha, _ := o.vm.HighAvailability()
+	if enabled, _ := ha.Enabled(); enabled {
+		return &enabled
+	}
+
+	return &enabled
+}
+
+func (o *OvirtMapper) mapMemory() *kubevirtv1.Memory {
+	// Guest memory
+	ovirtVMMemory, _ := o.vm.Memory()
+	guestMemory, _ := resource.ParseQuantity(strconv.FormatInt(ovirtVMMemory, 10))
+	memory := &kubevirtv1.Memory{}
+	memory.Guest = &guestMemory
+
+	// HugePages
+	hp := o.mapHugePages()
+	if hp != nil {
+		memory.Hugepages = hp
+	}
+
+	return memory
+}
+
+func (o *OvirtMapper) mapHugePages() *kubevirtv1.Hugepages {
+	var hugePageSize *string
+	if customProperties, available := o.vm.CustomProperties(); available {
+		for _, cp := range customProperties.Slice() {
+			if cpName, _ := cp.Name(); cpName == customPropertyHugePages {
+				cmValue, _ := cp.Value()
+				hugePageSize = &cmValue
+				break
+			}
+		}
+	}
+	if hugePageSize != nil {
+		return &kubevirtv1.Hugepages{
+			// HugePage size in oVirt custom_property is in MiB, so we add Mi suffix here.
+			PageSize: fmt.Sprintf("%sMi", *hugePageSize),
+		}
+	}
+
+	return nil
+}
+
+func (o *OvirtMapper) mapResourceRequirements() kubevirtv1.ResourceRequirements {
+	reqs := kubevirtv1.ResourceRequirements{}
+	memoryPolicy, _ := o.vm.MemoryPolicy()
+	maxMemory, _ := memoryPolicy.Max()
+	maxMemoryQuantity, _ := resource.ParseQuantity(strconv.FormatInt(maxMemory, 10))
+	reqs.Limits = map[corev1.ResourceName]resource.Quantity{
+		corev1.ResourceMemory: maxMemoryQuantity,
+	}
+
+	return reqs
+}
+
+// Graphical console is attached only in case the VM is not in headless mode
+func (o *OvirtMapper) mapGraphicalConsoles() *kubevirtv1.Devices {
+	// GraphicsConsole
+	vncEnabled := true
+	if gc, _ := o.vm.GraphicsConsoles(); len(gc.Slice()) == 0 {
+		vncEnabled = false
+	}
+	devices := &kubevirtv1.Devices{}
+	devices.AutoattachGraphicsDevice = &vncEnabled
+
+	// SerialConsole
+	if console, ok := o.vm.Console(); ok {
+		consoleEnabled, _ := console.Enabled()
+		devices.AutoattachSerialConsole = &consoleEnabled
+	}
+
+	return devices
+}
+
+func (o *OvirtMapper) mapLabels() map[string]string {
+	labels := map[string]string{}
+
+	// Origin
+	labels[LabelOrigin], _ = o.vm.Origin()
+
+	// Instance type
+	if instanceType, ok := o.vm.InstanceType(); ok {
+		labels[LabelInstanceType], _ = instanceType.Name()
+	}
+
+	// Tags
+	if tags, ok := o.vm.Tags(); ok {
+		var tagList []string
+		for _, tag := range tags.Slice() {
+			tagName, _ := tag.Name()
+			tagList = append(tagList, tagName)
+		}
+		labels[LabelTag] = strings.Join(tagList, ",")
+	}
+
+	// Fqdn
+	if fqdn, ok := o.vm.Fqdn(); ok {
+		labels[LabelFQDN] = fqdn
```

Comment:
should we map the `fqdn` to hostname field of  https://kubevirt.io/api-reference/v0.26.4/definitions.html#_v1_virtualmachineinstancespec ?

Reply:
Reading the doc of hostname field:

 https://kubevirt.io/user-guide/#/usage/dns?id=creating-unique-dns-entries-per-virtualmachineinstance

it's for internal resolution like <hostname>.<subdomain>. I think we can map it to hostname, so internal resolution will keep working, because if hostname is not specified it fallback to `vmi_name`.